tables_io/io_layer.py

```diff
@@ -274,6 +319,47 @@ def readHdf5ToTables(filepath):
     return tables
 
 
+def readHdf5ToDataframe(filepath):
+    """
+    Reads `pandas.DataFrame` objects from an hdf5 file.
+
+    Parameters
+    ----------
+    filepath: `str`
+        Path to input file
+
+    Returns
+    -------
+    df : `pandas.DataFrame`
+        The dataframe
+    """
+    if not HAS_HDF5:  #pragma: no cover
+        raise ImportError("h5py is not available, can't read hdf5")
+    if not HAS_PANDAS:  #pragma: no cover
+        raise ImportError("pandas is not available, can't read hdf5 to dataframe")
+    return pd.read_hdf(filepath)
```

Comment:
``pd.read_hdf`` doesn't save a normal HDF5 table, in the way that is expected by e.g. ``hdf5GroupToArrayDict``.  It saves some weird format.  e.g.:

```
>>> import pandas as pd
>>> df = pd.DataFrame([[1, 1.0, 'a']], columns=['x', 'y', 'z'])
>>> df.to_hdf('./store.h5', 'data')
src/tmp> h5ls -r ./store.h5 
/                        Group
/data                    Group
/data/axis0              Dataset {3}
/data/axis1              Dataset {1}
/data/block0_items       Dataset {1}
/data/block0_values      Dataset {1, 1}
/data/block1_items       Dataset {1}
/data/block1_values      Dataset {1, 1}
/data/block2_items       Dataset {1}
/data/block2_values      Dataset {1/Inf}
```

So right now the set of functions here is somewhat inconsistent and a round trip wouldn't really work.

Reply:
Yep.  This is also relevant to the comment below about hdf5 v. h5.  

Annoying, but so it goes.